tests/correctness/sync/sync-dot-product-opt.expect

```diff
@@ -0,0 +1,45 @@
+{
+  "in_0": [
+    4,
+    4,
+    6,
+    5,
+    10,
+    9,
+    16,
+    12,
+    0,
```

Comment:
Hm, can you add some zeros in the middle of the inputs as well to make sure the accelerator isn't just working for `0`s at the end?

Reply:
I added comments on the representation of sparse matrix at the beginning of sync-dot-product.futil